src/kbc_modules/online_sev_kbc/mod.rs

```diff
@@ -78,7 +79,16 @@ impl KbcInterface for OnlineSevKbc {
 
     async fn get_resource(&mut self, description: String) -> Result<Vec<u8>> {
         let desc = serde_json::from_str::<ResourceDescription>(description.as_str())?;
-        self.get_resource_from_kbs(desc.name).await
+        match ResourceName::from_str(desc.name.as_str())? {
```

Comment:
what happens if `desc.name` is just garbage? (none of the expected ResourceName enum values)

previously, this was passed to `get_resource_from_kbs()` which would then just send a query to the KBS with `secret_type="resource"` and `id="garbage"`.

now: `ResourceName::from_str("garbage")` might return Err and the KBS will not be queried.

this is a change in behaviour?

Another possible change:



maybe the definition of `ResourceDescription` should have field `name` with type `ResourceName` instead of `String`. this will have the effect but clearer declarations.

Reply:
Tweaked this slightly to preserve the current behavior although having a filter at the KBC might actually be a good idea.